__tests__/plugins.test.ts

```diff
@@ -0,0 +1,236 @@
+import { performance } from 'perf_hooks';
+
+import { jest } from '@jest/globals';
+import { number } from '@typeofweb/schema';
+
+import { createApp, createPlugin } from '../src';
+import { wait } from '../src/utils/utils';
+
+declare module '../src' {
+  interface TypeOfWebServerMeta {
+    readonly myPlugin: {
+      readonly someValue: number;
+      getUserById(id: string): Promise<readonly number[]>;
+    };
+  }
+}
+
+describe('plugins', () => {
+  it('should return function when cache is used', async () => {
+    const plugin = createPlugin('myPlugin', (_app) => {
+      return {
+        server(_server) {
+          return {
+            someValue: 42,
+            getUserById: {
+              cache: {
+                expireIn: 1,
+              },
+              fn: (id) => Promise.resolve(id.split('').map(Number)),
+            },
+          };
+        },
+      };
+    });
+    const app = createApp({}).route({
+      path: '/cache',
+      method: 'get',
+      validation: {},
+      handler: (request, _t) => {
+        return request.server.plugins.myPlugin.getUserById('123');
+      },
+    });
+    await app.plugin(plugin);
+
+    const result = await app.inject({
+      method: 'get',
+      path: '/cache',
+    });
+    expect(result.body).toEqual([1, 2, 3]);
+  });
+
+  it('should call the function only once when in cache', async () => {
+    const fn = jest.fn((id: string) => Promise.resolve(id.split('').map(Number)));
+
+    const plugin = createPlugin('myPlugin', (_app) => {
+      return {
+        server(_server) {
+          return {
+            someValue: 42,
+            getUserById: {
+              cache: {
+                expireAt: '00:00',
+              },
+              fn,
+            },
+          };
+        },
+      };
+    });
+
+    const app = createApp({}).route({
+      path: '/cache',
+      method: 'get',
+      validation: {},
```

Comment:
Maybe it would be a good idea, to make this field optional, I think it's better than passing empty object. 
But I also have to say that in real world app, we would probably just fill this field anyway.